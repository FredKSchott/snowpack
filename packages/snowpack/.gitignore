node_modules
pkg/*
!pkg/package.json
