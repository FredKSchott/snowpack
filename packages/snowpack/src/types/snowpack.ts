import type HttpProxy from 'http-proxy';
import {Plugin as RollupPlugin} from 'rollup';

export type DeepPartial<T> = {
  [P in keyof T]?: T[P] extends Array<infer U>
    ? Array<DeepPartial<U>>
    : T[P] extends ReadonlyArray<infer U>
    ? ReadonlyArray<DeepPartial<U>>
    : DeepPartial<T[P]>;
};

export type EnvVarReplacements = Record<string, string | number | true>;

export type SourceMappedCode = {code: string; map?: string};
export type SnowpackBuildMap = Record<string, string | SourceMappedCode>;

/** Standard file interface */
export interface SnowpackSourceFile {
  /** base extension (e.g. `.js`) */
  baseExt: string;
  /** file contents */
  contents: string;
  /** expanded extension (e.g. `.proxy.js` or `.module.css`) */
  expandedExt: string;
  /** if no location on disk, assume this exists in memory */
  locOnDisk: string;
}

export interface PluginLoadOptions {
  filePath: string;
  fileExt: string;
  isDev: boolean;
  isHmrEnabled: boolean;
}

export interface PluginTransformOptions {
  filePath: string;
  fileExt: string;
  contents: string | Buffer;
  isDev: boolean;
  isHmrEnabled: boolean;
}

export interface PluginRunOptions {
  isDev: boolean;
  isHmrEnabled: boolean;
}

/** map of extensions -> code (e.g. { ".js": "[code]", ".css": "[code]" }) */
export type PluginLoadResult = string | SnowpackBuildMap;

export interface PluginOptimizeOptions {
  buildDirectory: string;
}

export interface SnowpackPlugin {
  /** name of the plugin */
  name: string;
  /** Tell Snowpack how the load() function will resolve files. */
  resolve?: {
    /** file extensions that this load function takes as input (e.g. [".jsx", ".js", â€¦]) */
    input: string[];
    /** file extensions that this load function outputs (e.g. [".js", ".css"]) */
    output: string[];
  };
  /** load a file that matches resolve.input */
  load?(options: PluginLoadOptions): Promise<PluginLoadResult | null | undefined | void>;
  /** transform a file that matches resolve.input */
  transform?(options: PluginTransformOptions): Promise<string | null | undefined | void>;
  /** runs a command, unrelated to file building (e.g. TypeScript, ESLint) */
  run?(options: PluginRunOptions): Promise<unknown>;
  /** optimize the entire built application */
  optimize?(options: PluginOptimizeOptions): Promise<void>;
  /** Known dependencies that should be installed */
  knownEntrypoints?: string[];
}

export interface LegacySnowpackPlugin {
  defaultBuildScript: string;
  build?(options: PluginLoadOptions & {contents: string}): Promise<any>;
  bundle?(options: {
    srcDirectory: string;
    destDirectory: string;
    jsFilePaths: string[];
  }): Promise<any>;
}

/** Snowpack Build Plugin type */
export type SnowpackPluginFactory<PluginOptions = object> = (
  snowpackConfig: SnowpackConfig,
  pluginOptions?: PluginOptions,
) => SnowpackPlugin;

export type ProxyOptions = HttpProxy.ServerOptions & {
  // Custom on: {} event handlers
  on: Record<string, Function>;
};
export type Proxy = [string, ProxyOptions];

// interface this library uses internally
export interface SnowpackConfig {
  install: string[];
  extends?: string;
  exclude: string[];
  knownEntrypoints: string[];
  webDependencies?: {[packageName: string]: string};
  proxy: Proxy[];
  mount: Record<string, string>;
  alias: Record<string, string>;
  scripts: Record<string, string>;
  plugins: SnowpackPlugin[];
  devOptions: {
    secure: boolean;
    hostname: string;
    port: number;
    out: string;
    fallback: string;
    open: string;
    hmr: boolean;
  };
  installOptions: {
    dest: string;
    env: EnvVarReplacements;
    treeshake?: boolean;
    installTypes: boolean;
    sourceMap?: boolean | 'inline';
    externalPackage: string[];
    namedExports: string[];
    rollup: {
      plugins: RollupPlugin[]; // for simplicity, only Rollup plugins are supported for now
      dedupe?: string[];
    };
  };
  buildOptions: {
    baseUrl: string;
    webModulesUrl: string;
    clean: boolean;
    metaDir: string;
    minify: boolean;
    sourceMaps: boolean;
  };
  _extensionMap: Record<string, string>;
}

export interface CLIFlags extends Omit<Partial<SnowpackConfig['installOptions']>, 'env'> {
  help?: boolean; // display help text
  version?: boolean; // display Snowpack version
  reload?: boolean;
  config?: string; // manual path to config file
  env?: string[]; // env vars
  open?: string[];
  secure?: boolean;
}

export interface ImportMap {
  imports: {[packageName: string]: string};
}

export interface CommandOptions {
  cwd: string;
  config: SnowpackConfig;
  lockfile: ImportMap | null;
  pkgManifest: any;
}

/**
 * An install target represents information about a dependency to install.
 * The specifier is the key pointing to the dependency, either as a package
 * name or as an actual file path within node_modules. All other properties
 * are metadata about what is actually being imported.
 */
export type InstallTarget = {
  specifier: string;
  all: boolean;
  default: boolean;
  namespace: boolean;
  named: string[];
};

export type DependencyStats = {
  size: number;
  gzip: number;
  brotli?: number;
  delta?: number;
};

export type DependencyType = 'direct' | 'common';

export type DependencyStatsMap = {
  [filePath: string]: DependencyStats;
};

export type DependencyStatsOutput = Record<DependencyType, DependencyStatsMap>;
