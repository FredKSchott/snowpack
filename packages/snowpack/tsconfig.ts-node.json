{
  "ts-node": {
    "transpileOnly": true
  },
  "compilerOptions": {
    "target": "ES2015",
    "module": "CommonJS",
    "moduleResolution": "node",
    "esModuleInterop": true,
    "strict": true,
    "noImplicitAny": false,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "jsx": "react"
  },
  "include": ["src/**/*", "@types/**/*", "scripts/ts-node.bin.ts"],
  "exclude": ["node_modules"]
}
