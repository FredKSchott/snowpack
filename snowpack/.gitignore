lib/*
node_modules
vendor/generated/
vendor/types/
