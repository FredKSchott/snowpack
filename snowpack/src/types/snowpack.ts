import type HttpProxy from 'http-proxy';
import type * as http from 'http';
import type {InstallOptions} from 'esinstall';
import type {RawSourceMap} from 'source-map';

export type DeepPartial<T> = {
  [P in keyof T]?: T[P] extends Array<infer U>
    ? Array<DeepPartial<U>>
    : T[P] extends ReadonlyArray<infer U>
    ? ReadonlyArray<DeepPartial<U>>
    : DeepPartial<T[P]>;
};

export type SnowpackBuiltFile = {
  code: string | Buffer;
  map?: string;
};

export type SnowpackBuildMap = Record<string, SnowpackBuiltFile>;

/** Standard file interface */
export interface SnowpackSourceFile<Type = string | Buffer> {
  /** base extension (e.g. `.js`) */
  baseExt: string;
  /** file contents */
  contents: Type;
  /** expanded extension (e.g. `.proxy.js` or `.module.css`) */
  expandedExt: string;
  /** if no location on disk, assume this exists in memory */
  locOnDisk: string;
}

export interface PluginLoadOptions {
  filePath: string;
  fileExt: string;
  isDev: boolean;
  isSSR: boolean;
  isHmrEnabled: boolean;
}

export interface PluginTransformOptions {
  id: string;
  fileExt: string;
  contents: string | Buffer;
  isDev: boolean;
  isHmrEnabled: boolean;
}

export interface PluginRunOptions {
  isDev: boolean;
  isHmrEnabled: boolean;
}

/** map of extensions -> code (e.g. { ".js": "[code]", ".css": "[code]" }) */
export type PluginLoadResult = SnowpackBuildMap;

export type PluginTransformResult = {contents: string; map: string | RawSourceMap};

export interface PluginOptimizeOptions {
  buildDirectory: string;
}

export interface SnowpackPlugin {
  /** name of the plugin */
  name: string;
  /** Tell Snowpack how the load() function will resolve files. */
  resolve?: {
    /**
       file extensions that this load function takes as input (e.g. [".jsx",
       ".js", â€¦])
     */
    input: string[];
    /**
       file extensions that this load function outputs (e.g. [".js", ".css"])
     */
    output: string[];
  };
  /** load a file that matches resolve.input */
  load?(options: PluginLoadOptions): Promise<PluginLoadResult | string | null | undefined | void>;
  /** transform a file that matches resolve.input */
  transform?(
    options: PluginTransformOptions,
  ): Promise<PluginTransformResult | string | null | undefined | void>;
  /** runs a command, unrelated to file building (e.g. TypeScript, ESLint) */
  run?(options: PluginRunOptions): Promise<unknown>;
  /** optimize the entire built application */
  optimize?(options: PluginOptimizeOptions): Promise<void>;
  /** cleanup any long-running instances/services before exiting.  */
  cleanup?(): void | Promise<void>;
  /** Known dependencies that should be installed */
  knownEntrypoints?: string[];
  /** read and modify the Snowpack config object */
  config?(snowpackConfig: SnowpackConfig): void;
}

export interface LegacySnowpackPlugin {
  defaultBuildScript: string;
  build?(options: PluginLoadOptions & {contents: string | Buffer}): Promise<any>;
  bundle?(options: {
    srcDirectory: string;
    destDirectory: string;
    jsFilePaths: string[];
  }): Promise<any>;
}

/** Snowpack Build Plugin type */
export type SnowpackPluginFactory<PluginOptions = object> = (
  snowpackConfig: SnowpackConfig,
  pluginOptions?: PluginOptions,
) => SnowpackPlugin;

export type ProxyOptions = HttpProxy.ServerOptions & {
  // Custom on: {} event handlers
  on: Record<string, Function>;
};
export type Proxy = [string, ProxyOptions];

// interface this library uses internally
export interface SnowpackConfig {
  install: string[];
  extends?: string;
  exclude: string[];
  knownEntrypoints: string[];
  webDependencies?: {[packageName: string]: string};
  proxy: Proxy[];
  mount: Record<string, string>;
  alias: Record<string, string>;
  scripts: Record<string, string>;
  plugins: SnowpackPlugin[];
  devOptions: {
    secure: boolean;
    hostname: string;
    port: number;
    out: string;
    fallback: string;
    open: string;
    output: 'stream' | 'dashboard';
    hmr?: boolean;
    hmrDelay: number;
    hmrPort: number;
    hmrErrorOverlay: boolean;
  };
  installOptions: InstallOptions;
  buildOptions: {
    baseUrl: string;
    webModulesUrl: string;
    clean: boolean;
    metaDir: string;
    minify: boolean;
    sourceMaps: boolean;
    watch: boolean;
  };
  /** EXPERIMENTAL - This section is experimental and not yet finalized. May change across minor versions. */
  experiments: {
    /** (EXPERIMENTAL) If true, "snowpack build" should build your site for SSR. */
    ssr: boolean;
    /** (EXPERIMENTAL) Custom request handler for the dev server. */
    app?: (
      req: http.IncomingMessage,
      res: http.ServerResponse,
      next: (err?: Error) => void,
    ) => unknown;
  };
  _extensionMap: Record<string, string>;
}

export interface CLIFlags extends Omit<InstallOptions, 'env'> {
  help?: boolean; // display help text
  version?: boolean; // display Snowpack version
  reload?: boolean;
  config?: string; // manual path to config file
  env?: string[]; // env vars
  open?: string[];
  secure?: boolean;
  verbose?: boolean;
  quiet?: boolean;
}

export interface ImportMap {
  imports: {[packageName: string]: string};
}

export interface CommandOptions {
  cwd: string;
  config: SnowpackConfig;
  lockfile: ImportMap | null;
  pkgManifest: any;
}

export type LoggerLevel = 'debug' | 'info' | 'warn' | 'error' | 'silent'; // same as Pino
export type LoggerEvent = 'debug' | 'info' | 'warn' | 'error';
export interface LoggerOptions {
  /** (optional) change name at beginning of line */
  name?: string;
}
