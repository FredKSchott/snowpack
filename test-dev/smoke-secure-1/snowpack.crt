-----BEGIN CERTIFICATE-----
MIIERzCCAq+gAwIBAgIQUSVq6njLtPS7s+9BIHjGBzANBgkqhkiG9w0BAQsFADB1
MR4wHAYDVQQKExVta2NlcnQgZGV2ZWxvcG1lbnQgQ0ExJTAjBgNVBAsMHGFyYWxA
ZGV2LmFyLmFsIChBcmFsIEJhbGthbikxLDAqBgNVBAMMI21rY2VydCBhcmFsQGRl
di5hci5hbCAoQXJhbCBCYWxrYW4pMB4XDTIwMTIyMTE0MTcxMloXDTIzMDMyMTE0
MTcxMlowUDEnMCUGA1UEChMebWtjZXJ0IGRldmVsb3BtZW50IGNlcnRpZmljYXRl
MSUwIwYDVQQLDBxhcmFsQGRldi5hci5hbCAoQXJhbCBCYWxrYW4pMIIBIjANBgkq
hkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAzw5c6d2sxz7jWayEzyuZgod5bTW0VqQ8
I5WMuflQs/oiEX8OK9IPKkJ0lzDYe70jEbUlsnIPGP2XgyEwja1GLGqyt0BqdVsH
8vhmtBD0pICV5jA+R5v5o4jU07wsgHMC9rR4M5DPag3RjJWYsx7l5aosyEgMIpoF
t1r4XUukl2qFRlM8xXqywFyy0IfnT187ijh7VHT7Ibm+HHcHkFqbF9oqVUkR+NsB
0SEo4CEio5ufMCG0gmaqe4A81YGuLdUYYzPF5KOF4Hh1CAjXhKJr8dyETvixjZO3
40DmsguPbr5+xedof6USr0zuel56AA2Z3mwT0C8zndTd4AhpWx9GzQIDAQABo3gw
djAOBgNVHQ8BAf8EBAMCBaAwEwYDVR0lBAwwCgYIKwYBBQUHAwEwDAYDVR0TAQH/
BAIwADAfBgNVHSMEGDAWgBT2Ac1DO7MgFVDNm/l0GXsYfedtDzAgBgNVHREEGTAX
gglsb2NhbGhvc3SHBH8AAAGHBMCoAvkwDQYJKoZIhvcNAQELBQADggGBAEy0WzWB
ylaXs/VVFU5bXHCoVeSfMZlR0UOdIZ5CaAnzYx8YYVTKzR40hdI8w360BZkcPhcf
eESsx4lnHFgndvlIsrNRx5BxBVVRk6dzzgQMsQ3gr6auA+gCkzxNjXjB9eGwjVhU
rIyTQVDFhtAVq4rdBJd6dIipL8rlUuCXluVhBCcRfi3jiSPV2wNFvqGpPHN6X7yR
8j56zP7reSeL5CczdGbgJNLzp+jimz9NYLCuY97hURTTGeNV5+QGbgz18Uojjall
rgbWyrxIvf4SRxVOH2Y2htYtuinhB1UHQH8IuXpnBzoT1wAzT2tAxSy+Yi+QT5EK
kG0zVqqtYHkkHkBwnnGI0IA3hvQtvTsBCWZcQHM+1zK4BaVJJvtV60/Ejy8R1phw
2NhexVMUcLIrNNVZxS+O8l+Z5v0rBuL+rzuEB90idKWHCinJ0b4D9y7vUXdcXV6H
x4cunqRnZTYoHHZEY3YwKfHF5ICsToEKLDDgcmfJQZBEIepcmeyOtA/9sg==
-----END CERTIFICATE-----
