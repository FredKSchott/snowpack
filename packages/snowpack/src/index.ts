import * as colors from 'kleur/colors';
import path from 'path';
import yargs from 'yargs-parser';
import {addCommand, rmCommand} from './commands/add-rm';
import {command as buildCommand} from './commands/build';
import {command as devCommand} from './commands/dev';
import {command as installCommand} from './commands/install';
import {loadAndValidateConfig} from './config.js';
import {logger} from './logger';
import {CLIFlags} from './types/snowpack';
import {clearCache, readLockfile} from './util.js';

export {install as unstable_installCommand} from './commands/install';
export {createConfiguration} from './config.js';
export * from './types/snowpack';

const cwd = process.cwd();

function printHelp() {
  logger.info(
    `
${colors.bold(`snowpack`)} - A faster build system for the modern web.

  Snowpack is best configured via config file.
  But, most configuration can also be passed via CLI flags.
  ðŸ“– ${colors.dim('https://www.snowpack.dev/#configuration')}

${colors.bold('Commands:')}
  snowpack dev          Develop your app locally.
  snowpack build        Build your app for production.
  snowpack install      (Advanced) Install web-ready dependencies.

${colors.bold('Flags:')}
  --config [path]       Set the location of your project config file.
  --help                Show this help message.
  --version             Show the current version.
  --reload              Clear Snowpack's local cache (troubleshooting).
  --verbose             View debug info (where available)
  --quiet               Donâ€™t output anything (dev server will still log minimally)
    `.trim(),
  );
}

export async function cli(args: string[]) {
  // parse CLI flags
  const cliFlags = yargs(args, {
    array: ['install', 'env', 'exclude', 'externalPackage'],
  }) as CLIFlags;
  if (cliFlags.help) {
    printHelp();
    process.exit(0);
  }
  if (cliFlags.version) {
    logger.info(require('../package.json').version);
    process.exit(0);
  }
  if (cliFlags.reload) {
    logger.info(colors.yellow('! clearing cache...'));
    await clearCache();
  }

  // Load the current package manifest
  let pkgManifest: any;
  try {
    pkgManifest = require(path.join(cwd, 'package.json'));
  } catch (err) {
    logger.error(`package.json not found in directory: ${cwd}. Run \`npm init -y\` to create one.`);
    process.exit(1);
  }

  const cmd = cliFlags['_'][2];

  // Set this early -- before config loading -- so that plugins see it.
  if (cmd === 'build') {
    process.env.NODE_ENV = process.env.NODE_ENV || 'production';
  }
  if (cmd === 'dev') {
    process.env.NODE_ENV = process.env.NODE_ENV || 'development';
  }

  const commandOptions = {
    cwd,
    config: loadAndValidateConfig(cliFlags, pkgManifest),
    lockfile: await readLockfile(cwd),
    pkgManifest,
    logger,
  };

  if (cliFlags.verbose) logger.level = 'debug';
  if (cliFlags.quiet) logger.level = 'silent';

  if (cmd === 'add') {
    await addCommand(cliFlags['_'][3], commandOptions);
    return;
  }
  if (cmd === 'rm') {
    await rmCommand(cliFlags['_'][3], commandOptions);
    return;
  }

  if (cliFlags['_'].length > 3) {
    logger.error(`Unexpected multiple commands`);
    process.exit(1);
  }

  if (cmd === 'build') {
    await buildCommand(commandOptions);
    return;
  }
  if (cmd === 'dev') {
    await devCommand(commandOptions);
    return;
  }
  if (cmd === 'install' || !cmd) {
    await installCommand(commandOptions);
    return;
  }

  logger.error(`Unrecognized command: ${cmd}`);
  process.exit(1);
}
