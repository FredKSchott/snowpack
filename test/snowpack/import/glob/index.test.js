const {testFixture} = require('../../../fixture-utils');
const dedent = require('dedent');

const directory = {
  'deep/nest/globAbsolute.js': dedent`const modules = import.meta.glob('/pages/*.js');`,
  'deep/nest/globAlias.js': dedent`const modules = import.meta.glob('@root/pages/*.js');`,
  'pages/a.js': dedent`export default 'a'`,
  'pages/b.js': dedent`export default 'b'`,
  'pages/c.js': dedent`export default 'c'`,
  'glob.js': dedent`const modules = import.meta.glob('./pages/*.js');`,
  'globAbsolute.js': dedent`const modules = import.meta.glob('/pages/*.js');`,
  'globAlias.js': dedent`const modules = import.meta.glob('@root/pages/*.js');`,
  'globCommented.js': dedent`
    /* import.meta.glob('./pages/*.js'); */
    // import.meta.glob('./pages/*.js');
    
    const modules = null; // import.meta.glob('./pages/*.js');
    
    /* *** //
    * import.meta.glob('./pages/*.js');
    */
  `,
  'globEager.js': dedent`const modules = import.meta.globEager('./pages/*.js');`,
  'globSelf.js': dedent`const modules = import.meta.glob('./*.js');`,
  'snowpack.config.js': dedent`
    module.exports = {
      alias: {
        '@root': './',
      }
    }
  `,
};

describe('glob', () => {
  beforeAll(() => {
    // Needed until we make Snowpack's JS Build Interface quiet by default
    require('snowpack').logger.level = 'error';
  });

  it('Imports from directory', async () => {
    const result = await testFixture(directory);
    expect(result['glob.js']).toContain(`"./pages/a.js": () => import("./pages/a.js")`);
    expect(result['glob.js']).toContain(`"./pages/b.js": () => import("./pages/b.js")`);
    expect(result['glob.js']).toContain(`"./pages/c.js": () => import("./pages/c.js")`);
  });

  it('Imports from directory using absolute paths', async () => {
    const result = await testFixture(directory);
    expect(result['globAbsolute.js']).toContain(`"./pages/a.js": () => import("./pages/a.js")`);
    expect(result['globAbsolute.js']).toContain(`"./pages/b.js": () => import("./pages/b.js")`);
    expect(result['globAbsolute.js']).toContain(`"./pages/c.js": () => import("./pages/c.js")`);
  });

  it('Imports from directory using aliases', async () => {
    const result = await testFixture(directory);
    expect(result['globAlias.js']).toContain(`"./pages/a.js": () => import("./pages/a.js")`);
    expect(result['globAlias.js']).toContain(`"./pages/b.js": () => import("./pages/b.js")`);
    expect(result['globAlias.js']).toContain(`"./pages/c.js": () => import("./pages/c.js")`);
  });

  it('Imports from nested directory when using absolute paths', async () => {
    const result = await testFixture(directory);
    expect(result['deep/nest/globAbsolute.js']).toContain(
      `"../../pages/a.js": () => import("../../pages/a.js")`,
    );
    expect(result['deep/nest/globAbsolute.js']).toContain(
      `"../../pages/b.js": () => import("../../pages/b.js")`,
    );
    expect(result['deep/nest/globAbsolute.js']).toContain(
      `"../../pages/c.js": () => import("../../pages/c.js")`,
    );
  });

  it('Imports from nested directory when using aliases', async () => {
    const result = await testFixture(directory);
    expect(result['deep/nest/globAlias.js']).toContain(
      `"../../pages/a.js": () => import("../../pages/a.js")`,
    );
    expect(result['deep/nest/globAlias.js']).toContain(
      `"../../pages/b.js": () => import("../../pages/b.js")`,
    );
    expect(result['deep/nest/globAlias.js']).toContain(
      `"../../pages/c.js": () => import("../../pages/c.js")`,
    );
  });

  it('Statically imports from directory when using eagar', async () => {
    const result = await testFixture(directory);
    expect(result['globEager.js']).toContain(`import * as __glob__0_0 from './pages/a.js';`);
    expect(result['globEager.js']).toContain(`import * as __glob__0_1 from './pages/b.js';`);
    expect(result['globEager.js']).toContain(`import * as __glob__0_2 from './pages/c.js';`);
  });

  it('Ignores commented out module imports', async () => {
    const result = await testFixture(directory);
    expect(result['globCommented.js']).toContain(`const modules = null;`);
  });

  it('Does not import itself', async () => {
    const result = await testFixture(directory);
    expect(result['globSelf.js']).not.toContain(`globSelf.js`);
  });
});
