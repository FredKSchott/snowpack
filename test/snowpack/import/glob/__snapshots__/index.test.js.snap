// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`glob Imports various globs correctly 1`] = `
Object {
  "_snowpack/env.js": "export const MODE = \\"development\\";
export const NODE_ENV = \\"development\\";
export const SSR = false;",
  "deep/nest/globAbsolute.js": "import * as __SNOWPACK_ENV__ from '../../_snowpack/env.js';
import.meta.env = __SNOWPACK_ENV__;

const modules = {
	\\"../../pages/a.js\\": () => import(\\"../../pages/a.js\\"),
	\\"../../pages/b.js\\": () => import(\\"../../pages/b.js\\"),
	\\"../../pages/c.js\\": () => import(\\"../../pages/c.js\\")
};",
  "deep/nest/globAlias.js": "import * as __SNOWPACK_ENV__ from '../../_snowpack/env.js';
import.meta.env = __SNOWPACK_ENV__;

const modules = {
	\\"../../pages/a.js\\": () => import(\\"../../pages/a.js\\"),
	\\"../../pages/b.js\\": () => import(\\"../../pages/b.js\\"),
	\\"../../pages/c.js\\": () => import(\\"../../pages/c.js\\")
};",
  "glob.js": "import * as __SNOWPACK_ENV__ from './_snowpack/env.js';
import.meta.env = __SNOWPACK_ENV__;

const modules = {
	\\"./pages/a.js\\": () => import(\\"./pages/a.js\\"),
	\\"./pages/b.js\\": () => import(\\"./pages/b.js\\"),
	\\"./pages/c.js\\": () => import(\\"./pages/c.js\\")
};",
  "globAbsolute.js": "import * as __SNOWPACK_ENV__ from './_snowpack/env.js';
import.meta.env = __SNOWPACK_ENV__;

const modules = {
	\\"./pages/a.js\\": () => import(\\"./pages/a.js\\"),
	\\"./pages/b.js\\": () => import(\\"./pages/b.js\\"),
	\\"./pages/c.js\\": () => import(\\"./pages/c.js\\")
};",
  "globAlias.js": "import * as __SNOWPACK_ENV__ from './_snowpack/env.js';
import.meta.env = __SNOWPACK_ENV__;

const modules = {
	\\"./pages/a.js\\": () => import(\\"./pages/a.js\\"),
	\\"./pages/b.js\\": () => import(\\"./pages/b.js\\"),
	\\"./pages/c.js\\": () => import(\\"./pages/c.js\\")
};",
  "globCommented.js": "import * as __SNOWPACK_ENV__ from './_snowpack/env.js';
import.meta.env = __SNOWPACK_ENV__;

/* import.meta.glob('./pages/*.js'); */
// import.meta.glob('./pages/*.js');

const modules = null; // import.meta.glob('./pages/*.js');

/* *** //
* import.meta.glob('./pages/*.js');
*/",
  "globEager.js": "import * as __glob__0_0 from './pages/a.js';
import * as __glob__0_1 from './pages/b.js';
import * as __glob__0_2 from './pages/c.js';
import * as __SNOWPACK_ENV__ from './_snowpack/env.js';
import.meta.env = __SNOWPACK_ENV__;

const modules = {
	\\"./pages/a.js\\": __glob__0_0,
	\\"./pages/b.js\\": __glob__0_1,
	\\"./pages/c.js\\": __glob__0_2
};",
  "globSelf.js": "import * as __SNOWPACK_ENV__ from './_snowpack/env.js';
import.meta.env = __SNOWPACK_ENV__;

const modules = {
	\\"./glob.js\\": () => import(\\"./glob.js\\"),
	\\"./globAbsolute.js\\": () => import(\\"./globAbsolute.js\\"),
	\\"./globAlias.js\\": () => import(\\"./globAlias.js\\"),
	\\"./globCommented.js\\": () => import(\\"./globCommented.js\\"),
	\\"./globEager.js\\": () => import(\\"./globEager.js\\")
};",
  "pages/a.js": "export default 'a'",
  "pages/b.js": "export default 'b'",
  "pages/c.js": "export default 'c'",
}
`;
